src/disputables/scriptExtractors.js

```diff
@@ -0,0 +1,65 @@
+import environment from '../environment'
+import { getContract } from '../web3-contracts'
+import { performQuery, disputableVotingQuery } from './queries'
+import { sanitizeNetworkType, getNetworkType } from '../lib/web3-utils'
+
+import disputableDandelionVotingAbi from '../abi/disputables/DisputableDandelionVoting.json'
+import disputableDelayAbi from '../abi/disputables/DisputableDelay.json'
+
+const SUBGRAPH_NAME = environment('SUBGRAPH_NAME')
+
+export function delayExtractor(disputableAddress, disputableActionId) {
+  return extractFromContract(
+    disputableDelayAbi,
+    disputableAddress,
+    disputableActionId,
+    'delayedScripts',
+    4
+  )
+}
+
+export function dandelionVotingExtractor(
+  disputableAddress,
+  disputableActionId
+) {
+  return extractFromContract(
+    disputableDandelionVotingAbi,
+    disputableAddress,
+    disputableActionId,
+    'getVote',
+    10
+  )
+}
+
+export async function votingExtractor(disputableAddress, disputableActionId) {
+  // Disputable voting now saves the hash of the evmScript so we need to get it from the subgraph.
+  const networkType = sanitizeNetworkType(getNetworkType())
+  const subgraphUrl = `https://api.thegraph.com/subgraphs/name/aragon/aragon-dvoting-${networkType}${
+    SUBGRAPH_NAME ? `-${SUBGRAPH_NAME}` : ''
+  }`
+
+  const { data } = await performQuery(subgraphUrl, disputableVotingQuery, {
```

Comment:
I wonder if it would be a better API to hide `performQuery()` and only expose `performDisputableVotingQuery(id, voteId)`. Otherwise it might be a bit hard to know which params to use?

Reply:
Yes, good one!